[Tooling] Publish GitHub Releases for beta immediately

AliSoftware opened the pull request:
## Description

When the MC Release tool and automation builds a new beta or final build, it also creates a corresponding GitHub Release once the CI build is finished.

Currently those GitHub Releases are all created as draft.

 - This makes sense for the GitHub Release corresponding to the final build submitted to Google, because we only want to publish that GitHub Release (which has the side effect of creating the corresponding `git tag`) once we're sure that Google has approved the build (otherwise if we need to add last-minute commits to address the rejection from Google, we'd need to move the git tag after it had been created, which is generally a bad idea)
 - But for beta builds, we don't have this restriction, and it makes more sense to create the GitHub Pre-Releases for beta as published immediately rather than as Draft

The GitHub Prereleases being currently created as draft for betas has been an issue in the past because the Release Manager often forgets to publish them later. Automating that step should avoid that issue and stop us having old GitHub Prereleases left as Draft because we forgot to publish them.

AliSoftware committed:
Publish GitHub Releases for beta immediately

Instead of creating them as Draft Releases

While keeping the GitHub Releases for final builds as Draft still, because for those we only want to create the `git tag` after we are sure Google approved the build

AliSoftware committed:
Merge branch 'trunk' into tooling/publish-beta-github-releases

iangmaia commented on fastlane/Fastfile:
```diff
@@ -1150,7 +1150,8 @@ platform :android do
       version: app == WEAR_APP ? "#{version}w" : version,
       release_notes_file_path: METADATA_SOURCE_CHANGELOG_FILE_PATH[app],
-      prerelease: prerelease,
+      prerelease: prerelease, # Beta = Prerelease, Final = normal Release
```
💭  Not (necessarily) an issue, but it crossed my mind that on iOS we specifically mention the word "beta" as it is the lane parameter `options[:beta_release]`, while here we just always use `prerelease`.
At the same time, "beta" is how we call the build (and that's perhaps the track name in Google?), so it still makes sense.

I think my comment is more on the name of the original parameter and consistency between iOS/Android. Perhaps something to improve in the future (also, on iOS we don't have a separate lane / function to create the GH release as we do on Android).

AliSoftware replied:
Yeah the inconsistency annoyed me a bit as well, but it didn't feel worth it refactoring the `Fastfile` at the time (especially since I created this PR late yesterday evening 😅 ).
I'm ok to change it for consistency tho, I agree it would make more sense.

AliSoftware replied:
Addressed in https://github.com/woocommerce/woocommerce-android/pull/13094/commits/3870d7b88a53e0661fb166528ee6c87580c93d10

AliSoftware committed:
Rename `prerelease` parameter to `beta_release`

Especially now that this parameter is not just used for the `prerelease` parameter of the `create_github_release` action but also for its `is_draft` parameter

And also to be consistent with the parameter names used in WCiOS
